cedar-policy-core/src/entities.rs

```diff
@@ -104,6 +104,36 @@ impl Entities {
         self.entities.values()
     }
 
+    /// Adds the [`crate::ast::Entity`] to this [`Entities`].
+    /// Fails if the passed iterator contains any duplicate entities with this structure,
+    /// or if any error is encountered in the transitive closure computation.
+    ///
+    /// If you pass [`TCComputation::AssumeAlreadyComputed`], then the caller is
+    /// responsible for ensuring that TC and DAG hold before calling this method.
+    pub fn add_entities(
+        mut self,
+        collection: impl IntoIterator<Item = Entity>,
+        mode: TCComputation,
+    ) -> Result<Self> {
+        for entity in collection.into_iter() {
+            match self.entities.entry(entity.uid()) {
+                hash_map::Entry::Occupied(_) => return Err(EntitiesError::Duplicate(entity.uid())),
+                hash_map::Entry::Vacant(vacant_entry) => {
+                    vacant_entry.insert(entity);
+                }
+            }
+        }
+        match mode {
+            TCComputation::AssumeAlreadyComputed => (),
+            TCComputation::EnforceAlreadyComputed => {
+                enforce_tc_and_dag(&self.entities).map_err(Box::new)?
+            }
+            TCComputation::ComputeNow => compute_tc(&mut self.entities, true).map_err(Box::new)?,
+        };
+        self.evaluated_entities = None;
```

Comment:
should instead, we look to see if `evaluated_entities` is Some, and if it is, evaluate these new entities as well so that `evaluated_entities` remains Some?

Relatedly, what happens today if someone calls `.evaluate()` twice?

Reply:
The problem w/ doing that is that it introduces a new error mode, which means this method will have different error semantics from the normal constructor

Reply:
Makes sense, carry on